scripts/assets/file_content_validation_checks.textproto

```diff
@@ -339,6 +339,31 @@ file_content_checks {
   prohibited_content_regex: "name=.+(?<!_color).>@|name=.*[A-Z].*>@"
   failure_message: "All color declarations in component_color.xml and color_palette.xml should end with _color suffix following snake_case naming convention."
 }
+file_content_checks {
+  file_path_regex: "app/src/main/res/layout.*?/.+?\\.xml"
+  prohibited_content_regex: "@color/(?!component_color_).+"
+  failure_message: "Only colors from component_colors.xml may be used in layouts."
+}
+file_content_checks {
+  file_path_regex: "app/src/main/java/org/oppia/android/app.*?/.+ActivityPresenter\\.kt"
```

Comment:
You may be able to combine this & the others into a single check for Kotlin files. We also need to add View and ViewPresenter.

Reply:
Yes, I also thinking to combine them but to getting the pattern of path, how to write expression that enables for all of these?

Yes we need View ans ViewPresenter, But for View file "View" Present as prefix and ViewPresenter as suffix. So we need two checks for that?

Reply:
Done.

Reply:
Sorry I'm not sure that I understand the question for views--isn't it the case that all of our custom views end with 'View'? We don't currently have view presenters today as far as I know, but that's technically a pattern that's possible (and something we'll move toward once we introduce some of our planned code generation for UI code).